src/app/course-manager/manage-course/manage-course.component.ts

```diff
@@ -49,48 +58,39 @@ export class ManageCourseComponent implements OnInit {
 
   ngOnInit() {
     if (this.editCourseId) {
-      this.coursesService.getCourseById(this.editCourseId).subscribe(course => {
-        this.editCourse = course;
-        this.breadcrumbsService.setBreadcrumb(
-          this.breadcrumb,
-          this.editCourse.name
-        );
-        this.manageCourseForm.setValue({
-          name: this.editCourse.name,
-          description: this.editCourse.description.trim(),
-          length: this.editCourse.length,
-          date: this.datePipe.transform(this.editCourse.date, 'd MMM, yyyy'),
-          authors: this.editCourse.authors
-        });
+      this.editCourse$.subscribe(course => {
+        if (course) {
+          this.manageCourseForm.setValue({
+            name: course.name,
+            description: course.description.trim(),
+            length: course.length,
+            date: this.datePipe.transform(course.date, 'd MMM, yyyy'),
+            authors: course.authors
+          });
+        }
       });
     }
   }
 
   onSubmit() {
+    let newCourse: Course;
     if (this.editCourseId) {
-      this.coursesService
-        .updateCourse({
-          ...this.editCourse,
+      this.editCourse$.subscribe(course => {
+        newCourse = {
+          ...course,
           ...this.manageCourseForm.value,
           date: new Date(this.manageCourseForm.controls.date.value)
-        })
-        .subscribe(
-          () => {},
-          err => console.error(err)
-        );
+        };
```

Comment:
all state transformations and manipulations  should be done in reducers, that's their purpose, and it's easier to test there. In component you should dispatch only action with required params.